src/celengine/stardb.cpp

```diff
@@ -413,11 +416,18 @@ void StarDatabase::getStarName(const Star& star, char* nameBuffer, unsigned int
         StarNameDatabase::NumberIndex::const_iterator iter = namesDB->getFirstNameIter(catalogNumber);
         if (iter != namesDB->getFinalNameIter() && iter->first == catalogNumber)
         {
-            if (i18n && iter->second != _(iter->second.c_str()))
-                strncpy(nameBuffer, _(iter->second.c_str()), bufferSize);
-            else
-                strncpy(nameBuffer, iter->second.c_str(), bufferSize);
+            if (i18n)
+            {
+                const char * local = D_(iter->second.c_str());
+                if (iter->second != local)
+                {
+                    strncpy(nameBuffer, local, bufferSize);
```

Comment:
should it be strncpy(nameBuffer, local, bufferSize - 1) ?

Reply:
`strncpy` doesn't add the final '\0' if `bufferSize <= strlen(local)` so if you pass `bufferSize - 1` then `nameBuffer` is not guarantied to have final '\0'. That's why the next line is mandatory and why there is no difference whether we pass `nameBuffer` or`nameBuffer - 1` .